packages/backend/src/open_payments/payment/incoming_remote/service.ts

```diff
@@ -120,10 +120,20 @@ async function getGrant(
 
   if (existingGrant) {
     if (existingGrant.expired) {
-      // TODO https://github.com/interledger/rafiki/issues/795
-      const errorMessage = 'Grant access token expired'
-      deps.logger.error({ grantOptions }, errorMessage)
-      return RemoteIncomingPaymentError.ExpiredGrant
+      try {
+        const rotatedToken = await deps.openPaymentsClient.token.rotate({
+          url: existingGrant.managementUrl,
+          accessToken: existingGrant.accessToken
+        })
+        return deps.grantService.update(existingGrant, {
+          accessToken: rotatedToken.access_token.value,
+          managementUrl: rotatedToken.access_token.manage,
+          expiresIn: rotatedToken.access_token.expires_in
+        })
+      } catch (err) {
+        deps.logger.error({ grantOptions }, 'Grant token rotation failed.')
+        return RemoteIncomingPaymentError.ExpiredGrant
```

Comment:
I'm wondering if we should instead throw or if this should be renamed.
If I'm a GNAP/Open Payments-aware client and was told a relevant grant was expired, I would think "well, that grant's access token should be rotated". But rotation was already attempted and failed.

Reply:
that makes sense to me - I wouldn't mind logging out the error here in the log on 134, and just rethrowing the error. Since it's mostly an Admin API, we'll return a error message 500

Reply:
:+1: